### PR TITLE
[Integration][AWS-v3] Install new AWS-v3 integration

### PR DESCRIPTION
**Pull request by shariff-6**
### **User description**
# Description

What -

- Adds a new AWS v3 integration to Port Ocean, supporting both single and multi-account AWS environments.
- Implements modular, async authentication and session management strategies.
- Introduces region resolution, resource resync, and core utility modules.
- Provides comprehensive unit  for all new components.

Why -

- To enable scalable, maintainable, and robust AWS integration with modern async and multi-account capabilities.
- To improve performance, error handling, and flexibility in AWS resource synchronization.

How -

- Implements a strategy pattern for AWS session management, supporting both static credentials and STS assume-role flows.
- Adds region policy and resolver logic for fine-grained region selection and filtering.
- Refactors the integration entrypoint to use async generators and concurrency for efficient, batched resource syncing.
- Introduces core modules for async pagination, error handling, and protocol/type definitions.
- Expands test coverage with new fixtures and detailed unit tests for all auth, strategy, and utility modules.

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

___

### **PR Type**
Enhancement, Tests, Documentation

___

### **Description**
• Introduces a new AWS v3 integration with comprehensive async authentication and multi-account support
• Implements modular session management strategies using strategy pattern for both single and multi-account AWS environments
• Adds async CloudControl API synchronization with batch processing, region filtering, and concurrency controls
• Provides extensive unit test coverage for all authentication providers, session strategies, and core utilities
• Includes region resolution and policy-based filtering for fine-grained AWS resource selection
• Adds comprehensive project configuration with Poetry, SonarQube, and development tooling setup
• Implements Port Ocean integration with blueprints and resource mapping for AWS accounts and cloud resources
• Provides documentation including README, contributing guidelines, and changelog initialization

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>test_providers.py</strong><dd><code>Add comprehensive unit tests for AWS credential providers</code></dd></summary>
<hr>

integrations/aws-v3/tests/auth/test_providers.py

• Comprehensive unit tests for <code>StaticCredentialProvider</code> and <br><code>AssumeRoleProvider</code> classes<br> • Tests cover credential retrieval, session <br>creation, error handling, and various parameter combinations<br> • <br>Includes helper functions for mocking STS context managers and testing <br>different AWS regions<br> • Tests validation of refreshable credentials <br>behavior and assume role functionality

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-568706bf597908f1ba9373652c513581f31e110bd6ea5194389ef40e34fc7289">+458/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_strategies.py</strong><dd><code>Add unit tests for AWS session strategies and health checks</code></dd></summary>
<hr>

integrations/aws-v3/tests/auth/test_strategies.py

• Unit tests for <code>SingleAccountStrategy</code> and <code>MultiAccountStrategy</code> <br>session management<br> • Tests for health check functionality including <br>success/failure scenarios and concurrency control<br> • Validation of <br>account session generation and error handling for both single and <br>multi-account setups<br> • Tests for session creation with various <br>credential configurations and ARN validation

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-8d712305b9693a2789d3e3090051e45f5003a1a7ed004eb5bfd483668dfc9bdc">+414/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_session_factory.py</strong><dd><code>Add unit tests for AWS session factory and strategy creation</code></dd></summary>
<hr>

integrations/aws-v3/tests/auth/test_session_factory.py

• Tests for <code>ResyncStrategyFactory</code> creation logic for both single and <br>multi-account configurations<br> • Validation of strategy caching, health <br>check execution, and error handling<br> • Tests for <br><code>get_all_account_sessions</code> function with different account types and <br>configurations<br> • Comprehensive test coverage for session factory <br>initialization and strategy selection

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-207de3f8e1ae80168d09ca592be9ac0edcf38baa5a0360365b01d50e107f12c1">+293/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>conftest.py</strong><dd><code>Add comprehensive test fixtures for AWS integration testing</code></dd></summary>
<hr>

integrations/aws-v3/tests/conftest.py

• Comprehensive test fixtures for mocking AWS sessions, credentials, <br>and configurations<br> • Mock implementations for STS clients, account <br>clients, and various AWS service responses<br> • Test fixtures for both <br>single and multi-account configurations with proper credential mocking<br> <br>• Setup for Port Ocean context mocking to prevent initialization <br>errors during testing

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-cc93b12b2fe1577a9cf9ba0dce85aad5422898390bf0a81410fc45b2bc42d3e1">+215/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_utils.py</strong><dd><code>Add unit tests for AWS authentication utility functions</code>&nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/tests/auth/test_utils.py

• Unit tests for AWS utility functions including ARN normalization and <br>account extraction<br> • Tests for custom exception classes <br><code>AWSSessionError</code> and <code>CredentialsProviderError</code><br> • Validation of ARN <br>parsing functionality with different input types and edge cases<br> • <br>Tests for whitespace handling and mixed data type processing in ARN <br>lists

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-6150593627241a74f41fa4d869b9029d701f774ae4615d6c38839eb9d76a6a22">+140/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_region_resolver.py</strong><dd><code>Add unit tests for AWS region resolver functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/tests/auth/test_region_resolver.py

• Unit tests for <code>RegionResolver</code> class covering region discovery and <br>filtering<br> • Tests for AWS Account API integration to retrieve enabled <br>regions<br> • Validation of region policy enforcement and selector-based <br>region filtering<br> • Error handling tests for client creation failures <br>and empty API responses

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-105e06ad86cd0076cb48445f06e5c53da14305b9298ee170fc795f0db3784589">+147/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Add auth tests package initialization</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/tests/auth/__init__.py

• Empty init file for auth tests package<br> • Enables proper Python <br>package structure for test modules

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-dea027b6f20960c33737551c78d330045981bd2e9b46f1fa3e9a1f3a3abf9597">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>19 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>cloudcontrol_sync.py</strong><dd><code>Implement AWS CloudControl synchronization with async batch processing</code></dd></summary>
<hr>

integrations/aws-v3/aws/core/cloudcontrol_sync.py

• Core synchronization logic for AWS CloudControl API with async <br>resource processing<br> • Implements batch processing, region-based <br>resource syncing, and error handling for access denied scenarios<br> • <br>Supports both single and multi-account resource synchronization with <br>concurrency controls<br> • Includes global resource handling and region <br>filtering capabilities

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-831022309d59fc16f680987b4f4857e281553c48441d49bc64aa4d46aa121888">+291/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>multi_account_strategy.py</strong><dd><code>Implement multi-account AWS strategy with concurrent health checks</code></dd></summary>
<hr>

integrations/aws-v3/aws/auth/strategies/multi_account_strategy.py

• Implementation of multi-account AWS strategy with health check <br>capabilities<br> • Supports concurrent role assumption validation with <br>configurable batch processing<br> • Includes ARN validation, session <br>caching, and account information extraction<br> • Implements async <br>iterator for providing validated account sessions

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-cf29cfc48e3b0e42123a0265cf797d3a9caf6f5413d24507265aea6d66e82555">+144/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>paginator.py</strong><dd><code>Add async paginator for AWS API calls with batch processing</code></dd></summary>
<hr>

integrations/aws-v3/aws/core/paginator.py

• Async paginator implementation for AWS API calls with configurable <br>batch sizing<br> • Supports buffered resource processing to maintain <br>consistent batch sizes<br> • Includes logging and debugging capabilities <br>for pagination monitoring<br> • Handles AWS service metadata extraction <br>for account and region information

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-b5d2d61eb34226698b59feb2f35d7f0a5c56053c73751ae1f4ef5a2c4e8fa597">+112/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>integration.py</strong><dd><code>Add core AWS integration configuration and region policy models</code></dd></summary>
<hr>

integrations/aws-v3/integration.py

• Core integration configuration classes including <code>RegionPolicy</code> and <br><code>AWSDescribeResourcesSelector</code><br> • Implementation of region filtering <br>logic with allow/deny policies<br> • Integration with Port Ocean framework <br>through <code>AWSIntegration</code> and <code>AWSPortAppConfig</code><br> • Configuration models for <br>AWS resource selection and regional policies

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-324238a14fa85ac48ad533f27b52125db892ae7a842b2ee4343cbc5e916f30b0">+68/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>single_account_strategy.py</strong><dd><code>Implement single-account AWS strategy with credential validation</code></dd></summary>
<hr>

integrations/aws-v3/aws/auth/strategies/single_account_strategy.py

• Implementation of single-account AWS strategy with health check <br>validation<br> • Supports both explicit credentials and default AWS <br>credential chain<br> • Includes STS-based account validation and session <br>creation<br> • Provides async iterator for single account session <br>management

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-19ed0737a4d60ea82ec2892cd8d6a1173458f1a0a50b54e71113babc87f7c6f8">+67/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>assume_role_provider.py</strong><dd><code>Add AWS assume role provider with refreshable credentials</code></dd></summary>
<hr>

integrations/aws-v3/aws/auth/providers/assume_role_provider.py

• AWS assume role credential provider with refreshable credentials <br>support<br> • Implements STS assume role functionality with external ID <br>support<br> • Includes automatic credential refresh mechanism using <br><code>AioRefreshableCredentials</code><br> • Comprehensive error handling for role <br>assumption failures

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-ccb4c455ae150a720c2cceb24829f0741673829df2f8599094a7975d9a7821c5">+71/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>session_factory.py</strong><dd><code>Implement AWS session strategy factory with automatic selection</code></dd></summary>
<hr>

integrations/aws-v3/aws/auth/session_factory.py

• Factory pattern implementation for creating AWS session strategies<br> • <br>Automatic strategy selection based on configuration (single vs <br>multi-account)<br> • Strategy caching mechanism and health check execution <br>for single accounts<br> • Provides unified interface for accessing all <br>account sessions

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-524b95659492163d4c4a2de43c8337d05bd6ef72a812c1a0af1299e0ddda1db9">+66/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>utils.py</strong><dd><code>Add core AWS utility functions and type definitions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/aws/core/utils.py

• Core utility functions and type definitions for AWS integration<br> • <br>Exception detection utilities for access denied and resource not found <br>errors<br> • Global resource identification and JSON serialization helpers<br> <br>• Protocol definitions and configuration enums for CloudControl <br>operations

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-4fa843d3e06cdfbae3c3b1d8bf08c8fc44a88c959638d8643798ab0990b4572d">+74/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Add main AWS module exports and public API definition</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/aws/__init__.py

• Main module exports for AWS authentication and core functionality<br> • <br>Exposes all major classes and functions for external consumption<br> • <br>Organized imports for providers, strategies, utilities, and core <br>components<br> • Comprehensive <code>__all__</code> definition for clean API surface

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-d550ead255803c9ed8d67d1290d75dc4a62782759c7a716f7102b464bbce2ca8">+53/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Add main integration entry point with async resource synchronization</code></dd></summary>
<hr>

integrations/aws-v3/main.py

• Main integration entry point with Port Ocean framework integration<br> • <br>Implements async resource synchronization with account-level <br>concurrency control<br> • Uses semaphore-based concurrency limiting and <br>async iterator streaming<br> • Provides resync functionality for AWS <br>resources across multiple accounts

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-d379478674e394afda25614d88e760fe5782c90ae7d5fb34fa2b58f959a124f3">+45/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>static_provider.py</strong><dd><code>Add static AWS credential provider with boto3 chain fallback</code></dd></summary>
<hr>

integrations/aws-v3/aws/auth/providers/static_provider.py

• Static credential provider implementation for AWS authentication<br> • <br>Supports explicit credential specification or falls back to boto3 <br>credential chain<br> • Non-refreshable credential type suitable for <br>single-account scenarios<br> • Simple session creation with optional <br>credential attachment

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-3bb0de12b1529612f140ff46b86895f50f68d253ff566ac8c0ecf1c4add7db5e">+31/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>region_resolver.py</strong><dd><code>Add AWS region resolver with policy-based filtering</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/aws/auth/region_resolver.py

• AWS region discovery and filtering implementation<br> • Integrates with <br>AWS Account API to retrieve enabled regions<br> • Applies region policies <br>through selector-based filtering<br> • Supports both enabled and <br>enabled-by-default region statuses

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-93ea8739b99b71d9a166ef92132649c92760fee30f065e0804ea07f89d3b4640">+36/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Add core AWS module exports and API definitions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/aws/core/__init__.py

• Core module exports for AWS functionality<br> • Exposes synchronization <br>functions, paginator, and utility components<br> • Organized imports for <br>cloudcontrol sync, pagination, and utility functions<br> • Clean API <br>surface for core AWS integration components

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-07daf6ffad09a0d8cf92a8a7eba3accaa831f1e73b87f0fb2756099140fd1945">+39/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>utils.py</strong><dd><code>Add AWS authentication utilities and custom exceptions</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/aws/auth/utils.py

• Authentication utility functions and custom exception classes<br> • ARN <br>normalization and account extraction functionality<br> • Custom exceptions <br>for AWS session and credential provider errors<br> • Helper functions for <br>processing ARN lists and parsing

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-44c20972158a727832f72adf622f3a5e4071fc5acfe7e63a1ed11f18c20e63e0">+25/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>base.py</strong><dd><code>Add base credential provider abstract class and interfaces</code></dd></summary>
<hr>

integrations/aws-v3/aws/auth/providers/base.py

• Abstract base class for AWS credential providers<br> • Defines interface <br>for credential retrieval and session creation<br> • Type definitions for <br>different credential types (static, refreshable)<br> • Common <br>initialization pattern with configuration and session factory

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-28498902279bdb217ec7fe6abaeb1395b6c53be761426c224a059b01049a4a33">+27/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>base.py</strong><dd><code>Add base AWS session strategy and health check interfaces</code></dd></summary>
<hr>

integrations/aws-v3/aws/auth/strategies/base.py

• Abstract base classes for AWS session strategies and health check <br>functionality<br> • Defines interface for session creation and account <br>session iteration<br> • Separation of concerns between strategy <br>implementation and health checking<br> • Common patterns for strategy <br>initialization with providers and configuration

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-e097c2368f14a898cae33c57457054c4ed1e917902a129735ee96f8b33afa114">+29/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Add AWS authentication module exports and API definitions</code></dd></summary>
<hr>

integrations/aws-v3/aws/auth/__init__.py

• Authentication module exports for providers, strategies, and <br>utilities<br> • Exposes all authentication-related classes and functions<br> • <br>Organized imports for credential providers, session strategies, and <br>utilities<br> • Clean API surface for AWS authentication components

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-9be0aec1b9839ef70052e4d5010d4ba0f7c54f9809d6541d08b59345508ad125">+21/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Add AWS strategy module exports and API definitions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/aws/auth/strategies/__init__.py

• Strategy module exports for single and multi-account implementations<br> <br>• Exposes base strategy class and concrete implementations<br> • Clean API <br>surface for AWS session strategy components<br> • Organized imports for <br>strategy classes

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-45b67041829e68bae8d114e8eec859f78acf20bdea4aae3278898fb3cb732e92">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Add AWS provider module exports and API definitions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/aws/auth/providers/__init__.py

• Provider module exports for credential provider implementations<br> • <br>Exposes base provider class and concrete implementations (static, <br>assume role)<br> • Clean API surface for AWS credential provider <br>components<br> • Organized imports for provider classes

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-47107eab57b73a1f7502f464e2459e347c1e266a530e8e5de1df2eb0723efd0f">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>debug.py</strong><dd><code>Add debug entry point for integration development</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/debug.py

• Simple debug entry point for running the integration<br> • Uses Port <br>Ocean's run function for development and debugging<br> • Minimal <br>implementation for local testing and development

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-0583873f7c56e219b7d7e288b5dd886528699bd3e54caa5d21ea4887ceb4e592">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Add project configuration with dependencies and tooling setup</code></dd></summary>
<hr>

integrations/aws-v3/pyproject.toml

• Poetry configuration for AWS v3 integration project<br> • Dependencies <br>including Port Ocean, aiobotocore, and boto3 stubs<br> • Development <br>dependencies for testing, linting, and code formatting<br> • Configuration <br>for mypy, ruff, black, and pytest tools

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-1857eecaca5468790e0208fa126f43ead1d9eae72e692adfab519cdffa9e97d7">+118/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>blueprints.json</strong><dd><code>Add Port blueprints for AWS account and cloud resource entities</code></dd></summary>
<hr>

integrations/aws-v3/.port/resources/blueprints.json

• Port blueprints definition for AWS account and cloud resource <br>entities<br> • Schema definitions for AWS account properties (ARN, email, <br>status, joined method)<br> • Cloud resource blueprint with properties for <br>kind, tags, ARN, link, and region<br> • Relationship configuration between <br>cloud resources and AWS accounts

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-a955170f54d3ef406ab475b172f72e1a824e33d763f53936d3ab625361d6b4d9">+91/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>port-app-config.yml</strong><dd><code>Add Port app configuration for AWS resource mapping</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/.port/resources/port-app-config.yml

• Port app configuration for AWS resource mapping and synchronization<br> <br>• Resource definitions for S3 buckets, EC2 instances, and ECS clusters<br> <br>• Entity mappings with JQ expressions for property extraction and <br>transformation<br> • Configuration for using CloudControl API and resource <br>identification

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-4aed4c77464188e1d2f1907f1844f1833acd15ed5f118f4118fd2a80e810a517">+56/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>spec.yaml</strong><dd><code>Add integration specification with configuration parameters</code></dd></summary>
<hr>

integrations/aws-v3/.port/spec.yaml

• Integration specification with configuration parameters and features<br> <br>• Configuration options for account role ARN, external ID, and AWS <br>credentials<br> • Feature definition for cloud provider resource export<br> • <br>Sensitive parameter handling for security credentials

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-6c562d8345e86d384d1ade8eab077313727be91d3fc1fd21d9fae7c5bf45e0ba">+33/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>.env.example</strong><dd><code>Add environment configuration example file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/.env.example

• Environment variable examples for Port Ocean integration <br>configuration<br> • Template for Port client credentials and integration <br>settings<br> • Configuration for event listener type and resource <br>initialization

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-2423d2bb65c84faa274c718c6aa425afce126d4d09b353f263afe9ca19508ba5">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>Makefile</strong><dd><code>Add Makefile symlink for build infrastructure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/Makefile

• Symbolic link to shared infrastructure Makefile<br> • Enables common <br>build and development commands

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-c561edd5351ea961b5c685e2e24379a4fb77afacc47804530e7e4f4121efb04b">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>sonar-project.properties</strong><dd><code>Configure SonarQube project settings for code analysis</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/sonar-project.properties

• Configures SonarQube project settings for AWS v3 integration<br> • Sets <br>project key as <code>port-labs_ocean_aws-v3</code><br> • Associates with <code>port-labs</code> <br>organization

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-ca8f7cf283cfc2a25c467813f64c696fa868a460b6ef801dc472f312bd502d09">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>poetry.toml</strong><dd><code>Configure Poetry virtual environment settings</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/poetry.toml

• Configures Poetry virtual environment settings<br> • Enables virtual <br>environment creation in project directory<br> • Sets up local development <br>environment configuration

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-ffb359c61298e394f832fcf04f4b7660c95c6739a33aa5a7fb02350f39b5c793">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Add basic README documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/README.md

• Basic README with integration description and documentation links<br> • <br>Placeholder links for integration documentation and development guide<br> <br>• Brief description of AWS v3 integration purpose

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-97afd722d54826210a2ccc7b320dace1bf15b81ca6a44ca930679782c56261eb">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Initialize changelog for AWS v3 integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/CHANGELOG.md

• Creates a new changelog file for the AWS v3 integration<br> • Follows <br>Keep a Changelog format and Semantic Versioning standards<br> • Includes <br>placeholder for towncrier release notes

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-fe43d8fbfbf5549e68b0280f4c1fba9f31bf26332e99f16432e372b8bae2d5f9">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>CONTRIBUTING.md</strong><dd><code>Add contributing guidelines for AWS v3 integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

integrations/aws-v3/CONTRIBUTING.md

• Adds contributing guidelines for the AWS v3 integration<br> • Includes <br>placeholder for local development instructions<br> • Notes need for <br>specific setup instructions like credentials and rate limiting

</details>

  </td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-08b35840b17073579bdeb1735b5b9449ac7efc803f5bb3402904e679e39384f6">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>__init__.py</strong></td>
  <td><a href="https://github.com/port-labs/ocean/pull/1878/files#diff-067b158620926a66bc81e073a0b69e20b382e1bd2e31bcd741a26031c6543a8a">[link]</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: integrations/aws-v3/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,118 @@
+[tool.poetry]
+name = "aws-v3"
+version = "0.1.0-beta"
```
Comment (shalev007) [inline integrations/aws-v3/pyproject.toml]: Can we change it to dev instead of beta?
I don't want to it to appear to clients yet

File: integrations/aws-v3/.env.example
Comment (shalev007) [inline integrations/aws-v3/.env.example:1]: Nice
